setup.go

```diff
@@ -409,7 +409,12 @@ func setupBitswap(ctx context.Context, cfg Config, h host.Host, cr routing.Conte
 		provideEnabled = true
 		peerBlockRequestFilter = func(p peer.ID, c cid.Cid) bool {
 			_, ok := peers[p]
-			return ok
+			if !ok {
+				return false
+			}
+
+			has, err := bstore.Has(ctx, c)
+			return has && err == nil
```

Comment:
Help me understand what this fixes. It appears to make it so that if the s[erver does not have a block](https://github.com/ipfs/boxo/blob/main/bitswap/server/internal/decision/engine.go#L946-L949), then it should be treated as if access is denied to that block, [here](https://github.com/ipfs/boxo/blob/main/bitswap/server/internal/decision/engine.go#L771-L772), instead of treating that block as not found, [here](https://github.com/ipfs/boxo/blob/main/bitswap/server/internal/decision/engine.go#L781-L785).